src/mlmc/mlmc.py

```diff
@@ -12,6 +12,18 @@ def __init__(self, number_of_levels, sim_steps_range, sim, moments_object):
         :param sim_steps_range:     Simulations step fine and coarse
         :param sim:                 Instance of object Simulation
         :param moments_object:      Instance of moments object
+
+        JS TODO:
+
+        - Make 'sim_steps_range' part of simulation, an attribute: 'sim_param_range'
+          and allow it to be any pair of real positive parameters
+          Using logaritmic interpolatin we then compute parameters for individual levels and pass them to the simulation object.
+          So the meaning of the simulation parameter is simulation dependent.
+        - !! Code is wrong, sim is not instance of Simulation but SimulationSetting.
+          Original idea was that sim would be the simulation class however that makes setup of cofigurable simulations
+          problematic. Better would be to introduce a method Simulation.interpolate_by_precision(precision)
+          with t_level be a number between 0 and 1 given as (l/L) so this will interpolate simulation parameter and return copy
```

Comment:
Udělal bych to tak, že při vytváření levelu v mlmc by se konstruktoru předala simulace z Simulation.interpolate_by_precision(l/L), jemná simulace na předchozí úrovni a objekt pro výpočet momentů, takže by konstruktor vypadal -> (fine_sim, coarse_sim, moments_object). Sim_param_range předám Simulaci na úplném začátku při spuštění. Pokud jsem správně pochopil [flowSim](https://github.com/GeoMop/MLMC/pull/10/files#diff-bbe3c6fd65d1332d4f31194e96ac06edR47), tak siť se přidá simulaci vždy při vytvoření simulace, tzn. pro fine i coarse simulation nezávisle na sobě. Pak se fine_simulation předá [coarse_simulation](https://github.com/GeoMop/MLMC/pull/10/files#diff-bbe3c6fd65d1332d4f31194e96ac06edR102)  [Random array](https://github.com/GeoMop/MLMC/pull/10/files#diff-bbe3c6fd65d1332d4f31194e96ac06edR117), pak se vytvoří random array. [Zde](https://github.com/GeoMop/MLMC/pull/10/files#diff-bbe3c6fd65d1332d4f31194e96ac06edR125) self._input_sample je fine_sample?

Reply:
Asi bude lepsi mit pro simulaci dve tridy pro Simulaci, jednu generickou (FlowSimGeneric), ktera obsahuje data spolecna pro ruzne jemne simulace a metodu 'interpolate_by_precision' a 
pak konkretni simulaci (FlowSim), ktera uz v konstruktoru dostane i parameter presnosti a muze si tak udelat sit, je kompletne zadana.
... mozna to pak jeste pujde zjednodusit, ale zatim to nechme takto.

Reply:
Dalsi body:
- pro kazdy level mame dve simulace a vsechny jsou samostatne objekty, nicmene bylo by vhodne zajistit, ze fine_sim(l) ma stejnou sit jako coarse_sim(l+1). Pokud je nachame generovat si site samostatne tak ty site vetsinou nevyjdou uplne stejne, kvuli randomizaci. Asi by bylo vhodne do FlowSim pridat volitelny parametr 'mesh_file', kde by se dala predat jiz existujici sit.

- do fine_simulation se preda coarse_simulation pomoci 'set_coarse_sim', aby si to mohlo vytvorit spolecnou mnozinu bodu ve kterych se generuji korelovana pole pro fine i coarse

- pro jednotlie realizace se vola generovani pole v fime_simulation a jeho coarse cast se preda ven pomoci 'get_random_array' a  to se pak preda do coarse_sim pomoci 'set_random_array'  a 
vola se coarse_sim.cycle.

Reply:
Moznost predavani lambda funkce misto XYSimFactory:
class SimulationBase:
    ...
    @classmethod
    def make_sim(cls, config, sim_par_range, t_level):
        sim_par = SimulationBase.log_interpolation(sim_par_range, t_level)
        return cls(config, sim_par)

class FlowSimulation(SimulationBase):
   

sim_factory= lambda(t_level) : FlowSimulation.make_sim(my_config, my_range, t_level)
MLMC(..., sim_factory)